src/app/content/components/ContentExcerpt.tsx

```diff
@@ -30,22 +31,30 @@ const ContentExcerpt = (props: Props) => {
     forwardedRef,
     ...excerptProps
   } = props;
-
+  const services = useServices();
+  const [excerptSourceUrl, setExcerptSourceUrl] = React.useState('');
   const currentBook = assertDefined(useSelector(book), 'book not loaded');
   const sourcePage = typeof source === 'string'
     ? assertDefined(findArchiveTreeNodeById(currentBook.tree, source), 'page not found in book')
     : source;
 
-  const excerptSource = getBookPageUrlAndParams(
-    currentBook,
-    sourcePage
-  );
+  useEffect(() => {
+    const setBookPageUrlAndParams = async() => {
+      const { url } = await getBookPageUrlAndParams(
+        currentBook,
+        sourcePage,
+        services.bookConfigLoader
+      );
+      setExcerptSourceUrl(url);
+    };
+    setBookPageUrlAndParams();
+  });
 
   const fixedContent = React.useMemo(() => flow(
     addTargetBlankToLinks,
-    (newContent) => rebaseRelativeContentLinks(newContent, excerptSource.url),
-    (newContent) => resolveRelativeResources(newContent, excerptSource.url)
-  )(props.content), [props.content, excerptSource.url]);
+    (newContent) => rebaseRelativeContentLinks(newContent, excerptSourceUrl),
+    (newContent) => resolveRelativeResources(newContent, excerptSourceUrl)
+  )(props.content), [props.content, excerptSourceUrl]);
 
   return <div
```

Comment:
should probably not proceed with render until the url loads or the content could be invalid